src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -1821,4 +1821,95 @@ private final void appendChars(CharSequence s, int off, int end) {
         }
         count += end - off;
     }
+
+    /**
+     * Appends {@code count} copies of the character {@code c} to this sequence.
+     * <p>
+     * The length of this sequence increases by {@code count}.
+     *
+     * @param c      character to append
+     * @param count  number of times to copy
+     *
+     * @return  a reference to this object.
+     *
+     * @since 21
+     * @throws IllegalArgumentException  if {@code count} is less than zero
+     * @throws IndexOutOfBoundsException  if the result overflows the buffer
+     */
+    public AbstractStringBuilder repeat(char c, int count) {
+        if (count < 0) {
+            throw new IllegalArgumentException("count is less than zero: " + count);
+        } else if (count == 0) {
+            return this;
+        }
+        ensureCapacityInternal(this.count + count);
+        int index = this.count;
+        int limit = index + count;
+        boolean isLatin1 = isLatin1();
+        if (isLatin1 && StringLatin1.canEncode(c)) {
+            while (index < limit) {
+                value[index++] = (byte)c;
+            }
+        } else {
+            if (isLatin1) {
+                inflate();
+            }
+            while (index < limit) {
+                StringUTF16.putCharSB(value, index++, c);
+            }
+        }
+        this.count = limit;
+        return this;
+    }
+
+    /**
+     * Appends {@code count} copies of the specified {@code CharSequence} {@code cs}
+     * to this sequence.
+     * <p>
+     * The length of this sequence increases by {@code count} times the
+     * {@code CharSequence} length.
+     *
+     * @param cs     a {@code CharSequence}
+     * @param count  number of times to copy
+     *
+     * @return  a reference to this object.
+     *
+     * @since 21
+     * @throws IllegalArgumentException  if {@code count} is less than zero
+     * @throws IndexOutOfBoundsException  if the result overflows the buffer
+     */
+    public AbstractStringBuilder repeat(CharSequence cs, int count) {
+        if (count < 0) {
+            throw new IllegalArgumentException("count is less than zero: " + count);
+        } else if (count == 0) {
+            return this;
+        } else if (count == 1) {
+            return append(cs);
+        }
+        if (cs == null) {
+            cs = "null";
+        }
+        int length = cs.length();
+        if (length == 1) {
+            return repeat(cs.charAt(0), count);
+        }
+        int offset = this.count;
+        int valueLength = length << coder;
+        if ((Integer.MAX_VALUE - offset) / count < valueLength) {
+            throw new OutOfMemoryError("Required length exceeds implementation limit");
+        }
+        int total = count * length;
+        int limit = offset + total;
+        ensureCapacityInternal(limit);
```

Comment:
If I understand correctly, this may cause double reallocation, if inflation is necessary (string to spend is utf16 while current coder is latin1). Can this be optimized, at least for String/ASB inputs? Sorry if I'm missing something.

Reply:
I see. Well it's possible to create `ensureCapacityAndCoderInternal(capacity, coder)`, which handles both at once. It can be useful in other places as well. Probably this should be considered as a separate enhancement, outside of this PR.